dist-persist/wbstack/src/Info/GlobalSet.php

```diff
@@ -120,16 +123,18 @@ private static function getInfoFromApi( $requestDomain ) {
         curl_setopt($client, CURLOPT_HTTPHEADER, $headers);
         curl_setopt($client, CURLOPT_RETURNTRANSFER, true);
         curl_setopt( $client, CURLOPT_USERAGENT, "WBStack - MediaWiki - WBStackInfo::getInfoFromApi" );
-
+        
         $response = curl_exec($client);
+        if ($response === false) {
+            return new WBStackLookupFailure(502);
+        }
+        $responseCode = intval(curl_getinfo($client, CURLINFO_RESPONSE_CODE));
 
-        // TODO detect non 200 response here, and pass that out to the user as an error
-
-        $info = WBStackInfo::newFromJsonString($response, $requestDomain);
-        if (!$info) {
-            return null;
+        if ($responseCode > 399) {
+            return new WBStackLookupFailure($responseCode);
         }
-        return $info;
+
+        return WBStackInfo::newFromJsonString($response, $requestDomain);
```

Comment:
I *think* we still need to check what happens here; in the event this parsing fails and the static here returns a `null` we should build a WBStackLookupFailure; not quite sure how graceful or not failure would be in this case (since the logic in WBStackInfo is also very unpolished)

I tried this locally in a not super smart way as you can see at #411

Reply:
Yeah, good catch, we shall add this still. Will do so right away.

Reply:
@tarrow Added this in https://github.com/wbstack/mediawiki/pull/410/commits/75c7bd9505826fb41bd8abc6083af195a7fc156f